mono/metadata/attach.c

```diff
@@ -293,51 +289,44 @@ mono_attach_load_agent (MonoDomain *domain, char *agent, char *args, MonoObject
 	entry = mono_image_get_entry_point (image);
 	if (!entry) {
 		g_print ("Assembly '%s' doesn't have an entry point.\n", mono_image_get_filename (image));
-		g_free (agent);
-		return 1;
+		result = 1;
+		goto exit;
 	}
 
 	method = mono_get_method_checked (image, entry, NULL, NULL, error);
 	if (method == NULL){
 		g_print ("The entry point method of assembly '%s' could not be loaded due to %s\n", agent, mono_error_get_message (error));
-		mono_error_cleanup (error);
-		g_free (agent);
-		return 1;
+		result = 1;
+		goto exit;
 	}
 	
-	
-	main_args = (MonoArray*)mono_array_new_checked (domain, mono_defaults.string_class, (args == NULL) ? 0 : 1, error);
-	if (main_args == NULL) {
+	main_args = mono_array_new_handle (domain, mono_defaults.string_class, (args == NULL) ? 0 : 1, error);
+	if (MONO_HANDLE_IS_NULL (main_args)) {
 		g_print ("Could not allocate main method args due to %s\n", mono_error_get_message (error));
-		mono_error_cleanup (error);
-		g_free (agent);
-		return 1;
+		result = 1;
+		goto exit;
 	}
 
 	if (args) {
-		MonoString *args_str = mono_string_new_checked (domain, args, error);
+		MonoStringHandle args_str = mono_string_new_handle (domain, args, error);
 		if (!is_ok (error)) {
 			g_print ("Could not allocate main method arg string due to %s\n", mono_error_get_message (error));
-			mono_error_cleanup (error);
-			g_free (agent);
-			return 1;
+			result = 1;
+			goto exit;
 		}
-		mono_array_set_internal (main_args, MonoString*, 0, args_str);
+		MONO_HANDLE_ARRAY_SETREF (main_args, 0, args_str);
 	}
 
-
-	pa [0] = main_args;
-	mono_runtime_try_invoke (method, NULL, pa, exc, error);
+	pa [0] = MONO_HANDLE_RAW (main_args);
+	mono_runtime_try_invoke_handle (method, NULL_HANDLE, pa, error);
```

Comment:
This loss of `exc` parameter always bugs me.

Reply:
I think it is ok. Having both parameters is confusing. Changing code with both also confusing.

Reply:
The more I convert other code, the more I dislike that I removed this parameter.
I will add it back in another PR, and maybe this should get it too, for clarity.